src/routing/Router.svelte

```diff
@@ -61,6 +61,14 @@
           }
           store.setField('probe', newProbe);
 
+          // Scalar probes' names are snake_cased differently between the DB
+          // and Glean dictionary (see issue #1956 for more details).
+          // If this is a scalar probe then we'll need the probeId so we can
+          // snake_case it and derive a probe name whose casing matches the one in the DB.
+          if (newProbe.id && newProbe.id.startsWith('scalar/')) {
+            store.setField('probeId', newProbe.id.split('/')[1]);
+          }
```

Comment:
Looking at this again I think there might be an edge case where we click on a scalar probe, set the probeId, and then from that probe page, use the search bar, click on another probe that is not scalar, but the probeId from the last probe is still in the store which then can trigger the wrong probeName? This is very unlikely though because I believe we reset `newProbe` if we get new data so it should not be a problem?

Reply:
Yes, so I tested that and it picked up the right probe (second one, not scalar)

My understanding is that whenever a new probe is set, the router will call `useComponent`, which will run [reset `newProbe`](https://github.com/mozilla/glam/blob/98b11d3d9faec405b51e22976232886464dc309a/src/routing/Router.svelte#L54) so I think we're good there